### PR TITLE
Guides - Profiles API tutorial

### PR DESCRIPTION
**Pull request by raksiv**

### CODE REVIEW

**Review by jyecusch — CHANGES_REQUESTED** [review]

File: docs/guides/profiles.md [hunk 1]
```diff
@@ -0,0 +1,403 @@
...
+1. Use Nitric to create an API to create and update profiles
+2. Create handlers for the following API operations
+
+| **Method** | **Route**              | **Description**                  |
+| ---------- | ---------------------- | -------------------------------- |
+| `GET`      | /profiles/{$profileId} | Get a specific profile by its Id |
+| `GET`      | /profiles              | List all profiles                |
+| `POST`     | /profiles              | Create a new profile             |
+| `DELETE`   | /profiles/{$profileId} | Delete a profile                 |
+| `PUT`      | /profiles/{$profileId} | Update a profile                 |
```
[the remark this answers is not in the document]
Comment (raksiv) [reply docs/guides/profiles.md]: Updated to {id}

File: docs/guides/profiles.md [hunk 2]
```diff
@@ -0,0 +1,403 @@
...
+
+3. Running locally for testing
+4. Deploy to a cloud of your choice
+5. (Optional) Add handlers for the following API operations
+
+| **Method** | **Route**                    | **Description**                  |
+| ---------- | ---------------------------- | -------------------------------- |
+| `GET`      | /profiles/{id}/image/upload  | Get a profile image upload URL   |
+| `GET`      | profiles/{id}/image/download | Get a profile image download URL |
```
[the remark this answers is not in the document]
Comment (raksiv) [reply docs/guides/profiles.md:21]: Updated to same as above.

File: docs/guides/profiles.md [hunk 3]
```diff
@@ -0,0 +1,403 @@
...
+
+```bash
+nitric new
+```
+
+Create a project name and select the TypeScript template.
+
+```text
+? What is the name of the stack? my-profile-api
+? Choose a template: official/TypeScript - Starter
```
[the remark this answers is not in the document]
Comment (raksiv) [reply docs/guides/profiles.md:37]: Updated with new CLI output.

File: docs/guides/profiles.md [hunk 4]
```diff
@@ -0,0 +1,403 @@
...
+```
+
+The scaffolded project should have the following structure:
+
+```text
++--functions/
+|  +-- hello.ts
++--node_modules/
+|  ...
++--package.json
```
[the remark this answers is not in the document]
Comment (raksiv) [reply docs/guides/profiles.md:61]: Updated filepath to reintroduce nitric.yaml.

File: docs/guides/profiles.md [hunk 5]
```diff
@@ -0,0 +1,403 @@
...
+The template project is successfully scaffolded.
+
+## Coding our Profile API
+
+Let's start by initializing our profiles api, create a file named 'profiles.ts' within functions and add the following code.
+
+Here we are defining the following -
+
+- an API named public,
+- a collection named profiles with reading/writing permissions
```
[the remark this answers is not in the document]
Comment (raksiv) [reply docs/guides/profiles.md]: Done, reversed order for more clarity.

File: docs/guides/profiles.md [hunk 6]
```diff
@@ -0,0 +1,403 @@
...
+
+### Define our profile contents
+
+```typescript
+interface Profile {
+  name: string;
+  age: string;
+  homeTown: string;
+}
+```
```
[the remark this answers is not in the document]
Comment (raksiv) [reply docs/guides/profiles.md]: Removed the profile interface from use.

File: docs/guides/profiles.md [hunk 7]
```diff
@@ -0,0 +1,403 @@
...
+    output: await profiles.query().fetch(),
+  });
+});
+```
+
+### Remove profile with a delete method
+
+```typescript
+profileApi.delete('/profiles/:id', async (ctx) => {
+  const id = ctx.req.params['id'];
```
[the remark this answers is not in the document]
Comment (raksiv) [reply docs/guides/profiles.md]: Done, across all examples.

File: docs/guides/profiles.md [hunk 8]
```diff
@@ -0,0 +1,403 @@
...
+
+  try {
+    // Delete the profile from the collection
+    const profile = await profiles.doc(id).delete();
```
[the remark this answers is not in the document]
Comment (raksiv) [reply docs/guides/profiles.md]: Removed variable assignment and await

File: docs/guides/profiles.md [hunk 9]
```diff
@@ -0,0 +1,403 @@
...
+### Retrieve profile with a get method
+
+```typescript
+profileApi.get('/profiles/:id', async (ctx) => {
+  const id = ctx.req.params['id'];
+
+  try {
+    // Get the profile from our collection
+    const profile = await profiles.doc(id).get();
+    return ctx.res.json(profile);
```
[the remark this answers is not in the document]
Comment (raksiv) [reply docs/guides/profiles.md:145]: Removed return.

**Review by jyecusch — APPROVED** [review]

**Review by tjholm — APPROVED** [review]

**Pull request merged**